### DISCUSSION TITLE
Template for sonoff-tx ultimate 2 relay for covers

### DISCUSSION THREAD

**Discussion by kroon040** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#4] [Q&A] [answered]
Is it possible to have the config made for to use with covers. It should be nice to use it swipe left to right closes and from right to left opens and stop multitouch, and some led colors them moving.

Any idea for a config|?

**Comment by danielscheidler** [comment]
Yes, Cover ist the next Thing I will do.
I want to do Open/Close on the Touch Pads and Open/Close my curtains with swipe. But I will think about how I can realize this as much configurable as possible.

**Comment by kroon040** [comment]
Great. Please also with interlock (that 2 relays cannot be active same time) and also the stop with multitouch or touch so you can stop closing when you want.

**Comment by danielscheidler** [comment]
for sure with interlock. Don't want to burn the motors ;)
What about stop by default touch if cover is opening/closing? So you have not to think about another touch type. 

```
    on_press:
      then:
        - lambda: |
            if (id(my_cover).current_operation == COVER_OPERATION_IDLE) {
              id(my_cover).close();
            } else {
              id(my_cover).stop();
            }
```

**Reply by kroon040** [reply]
Ok that's fine

**Reply by kroon040** [reply]
maybe nice if the covers go up, the led animation with colors is scrolling up , and when down the leds scrolling down

**Reply by danielscheidler** [reply]
I thought about to show the position of the cover by the leds on one side. 
But an additional animation in the moving direction is also possible.

I think I can use my stairlight for this ;) 
https://github.com/SmartHome-yourself/rgb-treppenbeleuchtung-for-esphome

**Comment by kroon040** [comment]
Yes thats also nice. Than leds on if up and led off if down for position, and animation of led is maybe not needed, because when moving the leds also go live for position, or only when it's stopped moving?

**Reply by danielscheidler** [reply]
first progress for cover is done ;)

https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/assets/705724/74b127ee-8893-47ff-b0d3-4584580cfef7

**Comment by kroon040** [comment]
Great job, looks very nice already. I hope  this weekend to have time to flash the device, so if you have than a configuration that you can share.

**Reply by danielscheidler** [reply, marked as answer]
Sorry for the long delay. But cover config is now added to the repo including the audio-fix, visualize cover-position and changed effects for swipe.
But I choosed to control over buttons für up/down.
You can copy the full cover.yaml to change it to your needs. I think you only have to move the actions in the binary_sensors from touch to swipe.

**Comment by kroon040** [comment]
Great. I go try it this weekend. For the time I had flashed it with tasmota
